Makefile's prerequisites problem when 'make -j >1 build'

Hi,

I was preparing a package for AUR (Arch Linux) and my system has `-j` enabled for building packages. When I build the package (makepkg), I get some errors. If I add `-j 1` I can compile it without errors.

The error:
```
$ make -j 2 build
[ ! -d "/home/liam/Documents/Development/aur/nextcloud-spreed-signaling/src/nextcloud-spreed-signaling-v0.2.0/.git/hooks" ] || ln -sf "/home/liam/Documents/Development/aur/nextcloud-spreed-signaling/src/nextcloud-spreed-signaling-v0.2.0/scripts/pre-commit.hook" "/home/liam/Documents/Development/aur/nextcloud-spreed-signaling/src/nextcloud-spreed-signaling-v0.2.0/.git/hooks/pre-commit"
GOPATH="/home/liam/Documents/Development/aur/nextcloud-spreed-signaling/src/nextcloud-spreed-signaling-v0.2.0/vendor:/home/liam/Documents/Development/aur/nextcloud-spreed-signaling/src/nextcloud-spreed-signaling-v0.2.0" /usr/bin/go get github.com/rogpeppe/godeps
PATH=/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/opt/android-sdk/tools:/opt/android-sdk/tools/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/opt/pscripts/uihelper:/opt/pscripts/sysadmin GOPATH="/home/liam/Documents/Development/aur/nextcloud-spreed-signaling/src/nextcloud-spreed-signaling-v0.2.0/vendor:/home/liam/Documents/Development/aur/nextcloud-spreed-signaling/src/nextcloud-spreed-signaling-v0.2.0" ./vendor/bin/easyjson -all src/signaling/api_signaling.go
PATH=/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/opt/android-sdk/tools:/opt/android-sdk/tools/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/opt/pscripts/uihelper:/opt/pscripts/sysadmin GOPATH="/home/liam/Documents/Development/aur/nextcloud-spreed-signaling/src/nextcloud-spreed-signaling-v0.2.0/vendor:/home/liam/Documents/Development/aur/nextcloud-spreed-signaling/src/nextcloud-spreed-signaling-v0.2.0" ./vendor/bin/easyjson -all src/signaling/api_backend.go
PATH=/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/opt/android-sdk/tools:/opt/android-sdk/tools/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/opt/pscripts/uihelper:/opt/pscripts/sysadmin GOPATH="/home/liam/Documents/Development/aur/nextcloud-spreed-signaling/src/nextcloud-spreed-signaling-v0.2.0/vendor:/home/liam/Documents/Development/aur/nextcloud-spreed-signaling/src/nextcloud-spreed-signaling-v0.2.0" ./vendor/bin/easyjson -all src/signaling/api_proxy.go
/bin/sh: ./vendor/bin/easyjson: No such file or directory
make: *** [Makefile:95: src/signaling/api_signaling_easyjson.go] Error 127
make: *** Waiting for unfinished jobs....
/bin/sh: ./vendor/bin/easyjson: No such file or directory
make: *** [Makefile:95: src/signaling/api_backend_easyjson.go] Error 127
/bin/sh: ./vendor/bin/easyjson: No such file or directory
make: *** [Makefile:95: src/signaling/api_proxy_easyjson.go] Error 127
```

I didn't go through that much, but I believe the problem starts at line 108 of Makefile, `server: dependencies common`. When running multithreaded it will run `common` _at same time_ as `dependencies`. I might be completely wrong.

I suggest, in mean time, adding a warning on readme to not run with `-j >1`.

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -39,8 +39,10 @@ jobs:
 
     - name: Build applications
       run: |
-        make client
-        make server
+        echo "Building with $(nproc) threads"
+        make client -j$(nproc)
+        make proxy -j$(nproc)
+        make server -j$(nproc)
 
     - name: Run tests
       run: |

--- a/Makefile
+++ b/Makefile
@@ -99,26 +99,26 @@ coverhtml: dependencies vet common
 	sed -i "/_easyjson/d" cover.out && \
 	GOPATH=$(GOPATH) $(GO) tool cover -html=cover.out -o coverage.html
 
-%_easyjson.go: %.go
+%_easyjson.go: %.go easyjson
 	PATH=$(shell dirname $(GO)):$(PATH) GOPATH=$(GOPATH) ./vendor/bin/easyjson -all $*.go
 
-common: easyjson \
+common: \
 	src/signaling/api_signaling_easyjson.go \
 	src/signaling/api_backend_easyjson.go \
 	src/signaling/api_proxy_easyjson.go \
 	src/signaling/natsclient_easyjson.go \
 	src/signaling/room_easyjson.go
 
-client: dependencies common
+$(BINDIR):
 	mkdir -p $(BINDIR)
+
+client: dependencies common $(BINDIR)
 	GOPATH=$(GOPATH) $(GO) build $(BUILDARGS) -ldflags '$(INTERNALLDFLAGS)' -o $(BINDIR)/client ./src/client/...
 
-server: dependencies common
-	mkdir -p $(BINDIR)
+server: dependencies common $(BINDIR)
 	GOPATH=$(GOPATH) $(GO) build $(BUILDARGS) -ldflags '$(INTERNALLDFLAGS)' -o $(BINDIR)/signaling ./src/server/...
 
-proxy: dependencies common
-	mkdir -p $(BINDIR)
+proxy: dependencies common $(BINDIR)
 	GOPATH=$(GOPATH) $(GO) build $(BUILDARGS) -ldflags '$(INTERNALLDFLAGS)' -o $(BINDIR)/proxy ./src/proxy/...
 
 clean:

